eth/client.go

```diff
@@ -162,46 +156,35 @@ func NewClient(accountAddr ethcommon.Address, keystoreDir string, eth *ethclient
 	if err != nil {
 		return nil, err
 	}
+	backend.SetMaxGasPrice(maxGasPrice)
 
 	am, err := NewAccountManager(accountAddr, keystoreDir, signer)
 	if err != nil {
 		return nil, err
 	}
 
-	return &client{
-		accountManager: am,
-		backend:        backend,
-		controllerAddr: controllerAddr,
-		txTimeout:      txTimeout,
-	}, nil
-}
+	if err := am.Unlock(password); err != nil {
+		return nil, err
+	}
 
-func (c *client) Setup(password string, gasLimit uint64, gasPrice *big.Int) error {
-	err := c.accountManager.Unlock(password)
+	// Setting gasLimit and gasPrice to zero, nil allows it to be overwritten by eth.Backend
+	opts, err := am.CreateTransactOpts(0, nil)
```

Comment:
Since we're following the deprecation procedure described in the previous round of review, let's keep the existing functionality for -gasPrice and -gasLimit until they are actually removed?

Reply:
While I think we should be aware that by keeping the gasPrice flag around with existing behaviour it can lead to worse UX for the node however since the new behaviour is different I'm okay with keeping it around.